src/rulesets/LinearAlgebra/dense.jl

```diff
@@ -394,3 +394,51 @@ function rrule(
     end
     return Ω, lyap_pullback
 end
+
+#####
+##### `kron`
+#####
+
+function frule((_, Δx, Δy), ::typeof(kron), x::AbstractVecOrMat{<:Number}, y::AbstractVecOrMat{<:Number})
+    return kron(x, y), kron(Δx, y) + kron(x, Δy)
+end
+
+function rrule(::typeof(kron), x::AbstractVector{<:Number}, y::AbstractVector{<:Number})
+    function kron_pullback(z̄)
+        dz = reshape(unthunk(z̄), length(y), length(x))
+        x̄ = @thunk conj.(dz' * y)
```

Comment:
These need a projection using `ProjectTo`

Reply:
Ok, I'll take your word for it..

Reply:
How would we write a test that would catch this missing `ProjectTo`?

Reply:
Probably a test using a type that has structural zeros, e.g. a one-hot vector. Not certain if a sparse type on the stdlib would work or if you'd need to add your own.

Reply:
Actually, a simple test of `ProjectTo` is to make one arg real and one arg complex. Projection will real-ify the cotangent for the real argument, and the test would otherwise fail without it.